src/local_scheduler/local_scheduler_algorithm.h

```diff
@@ -307,6 +314,67 @@ int reconstruct_object_timeout_handler(event_loop *loop,
 void print_worker_info(const char *message,
                        SchedulingAlgorithmState *algorithm_state);
 
+/**
+ * Get the number of tasks, per actor handle, that have been executed on an
+ * actor so far.
+ *
+ * @param algorithm_state State maintained by the scheduling algorithm.
+ * @param actor_id The ID of the actor whose task counters are returned.
+ * @return A map from handle ID to the number of tasks submitted by that handle
+ *         that have executed so far.
+ */
+std::unordered_map<ActorHandleID, int64_t, UniqueIDHasher>
+get_actor_task_counters(SchedulingAlgorithmState *algorithm_state,
+                        ActorID actor_id);
+
+/**
+ * Set the number of tasks, per actor handle, that have been executed on an
+ * actor so far. All previous counts will be overwritten.
+ *
+ * @param algorithm_state State maintained by the scheduling algorithm.
+ * @param actor_id The ID of the actor whose task counters are returned.
+ * @param task_counters A map from handle ID to the number of tasks submitted
+ *        by that handle that have executed so far.
+ * @return Void.
+ */
+void set_actor_task_counters(
+    SchedulingAlgorithmState *algorithm_state,
+    ActorID actor_id,
+    const std::unordered_map<ActorHandleID, int64_t, UniqueIDHasher>
+        &task_counters);
+
+/**
+ * Get the actor's frontier of task dependencies.
```

Comment:
Since "frontier" is a new concept introduced in this PR -- and primarily abstracted away by these two functions? -- maybe add a clear explanation/intuition on what it is before these two functions?

Reply:
Yes, that a good idea!